packages/xstate-wallet/src/store/channel-store-entry.ts

```diff
@@ -229,14 +239,38 @@ export class ChannelStoreEntry {
 
     this.clearOldStates();
 
+    this.checkInvariants();
+
     return this.state(entry);
   }
 
+  private checkInvariants() {
```

Comment:
Should we call this in the constructor? I didn't notice that we had tests explicitly defining the order and that caught me up when I was trying to figure out what was wrong.